src/init.c

```diff
@@ -290,18 +290,20 @@ void segv_handler(int sig, siginfo_t *info, void *context)
         sigprocmask(SIG_UNBLOCK, &sset, NULL);
         jl_throw(jl_memory_exception);
     }
+    else if ( (sig == SIGSEGV && in_jl_) ||
```

Comment:
This doesn't look like a valid expression. I suggest adding:

#ifndef SEGV_EXCEPTION
#define SEGV_EXCEPTION (0)
#endif

Then you can use this as a variable rather than as a ifdef test, which generally makes the code easier to read and ensures that all code paths compile.